certification/internal/policy/container/has_license.go

```diff
@@ -2,32 +2,67 @@ package container
 
 import (
 	"fmt"
-	"strings"
+	"io/fs"
+	"os"
+	"path/filepath"
 
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
 	log "github.com/sirupsen/logrus"
 )
 
 const (
 	licensePath         = "/licenses"
 	newLine             = "\n"
 	minLicenseFileCount = 1
-	//size format string for find command, c denotes bytes
 )
 
 // HasLicenseCheck evaluates that the image contains a license definition available at
 // /licenses.
 type HasLicenseCheck struct{}
 
 func (p *HasLicenseCheck) Validate(imgRef certification.ImageReference) (bool, error) {
-	// TODO Implement
-	return false, fmt.Errorf("unimplemented in migration path to crane engine")
+	licenseFileList, err := p.getDataToValidate(imgRef.ImageFSPath)
+	if err != nil {
+		return false, err
+	}
+	return p.validate(licenseFileList)
+}
+
+func (p *HasLicenseCheck) getDataToValidate(mountedPath string) ([]fs.DirEntry, error) {
+	fullPath := filepath.Join(mountedPath, licensePath)
+	fileinfo, err := os.Stat(fullPath)
+	if err != nil {
+		log.Error(fmt.Sprintf("Error when checking for %s : ", licensePath), err)
+		return nil, err
+	}
+	if !fileinfo.IsDir() {
+		log.Error(fmt.Sprintf("%s is not a directory", licensePath))
+		return nil, errors.ErrLicensesNotADir
+	}
+
+	files, err := os.ReadDir(fullPath)
+	if err != nil {
+		log.Error(fmt.Sprintf("Error when reading directory %s", licensePath), err)
+		return nil, err
+	}
+	return files, nil
 }
 
-func (p *HasLicenseCheck) validate(licenseFileList string) (bool, error) {
-	licenseFileCount := strings.Count(licenseFileList, newLine)
-	log.Infof("%d Licenses found", licenseFileCount)
-	return licenseFileCount >= minLicenseFileCount, nil
+func (p *HasLicenseCheck) validate(licenseFileList []fs.DirEntry) (bool, error) {
+	nonZeroLength := false
+	for _, f := range licenseFileList {
+		info, err := f.Info()
+		if err != nil {
+			continue
+		}
```

Comment:
if the intention is to ignore the errors, can we update the above line to `info, _ := f.Info()`?

Reply:
That's not just ignoring the error. It keeps the line after the block (info.Size()) from executing. It's assuming that if there is an error, the info return value is invalid.

Reply:
BTW, if the err is nil, doesn't the Info() method return nil for the value of `FileInfo` which will automatically escape the following if block?

Reply:
I'm fine with leaving this as explicit as it is. I'm not seeing a gain in readability. Whether err is nil has nothing to do with the what the size of the file is.